jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt

```diff
@@ -4,148 +4,63 @@
-                LOG.info { "$it does not exist, won't watch" }
-                false
-            }
-        }.map {
-            val watchLocation = if (it.isFile()) it.parent else it
-            watchLocation.register(
-                watchService,
-                StandardWatchEventKinds.ENTRY_CREATE,
-                StandardWatchEventKinds.ENTRY_DELETE,
-                StandardWatchEventKinds.ENTRY_MODIFY
-            ) to watchLocation
-        }.forEach {
-            LOG.info { "Watching ${it.second} for file changes" }
-            watchKeys[it.first] = it.second
-        }
-
-        // Only start if not already running
-        if (watchFuture == null || watchFuture?.isDone == true) {
-            watchFuture = executor.submit(this::watch)
-        }
-    }
-
-    private fun watch() {
-        while (true) {
-            val key = try {
-                watchService.take()
-            } catch (e: InterruptedException) {
-                return
-            }
-
-            var invokeListeners = false
-            key.pollEvents().forEach {
-                try {
-                    val kind = it.kind()
-                    when (kind) {
-                        StandardWatchEventKinds.OVERFLOW -> {
-                            LOG.debug { "ProfileWatcher got an OVERFLOW" }
-                        }
-                        else -> {
-                            // Context path is relative to base registered to WatchKey
-                            val context = it.context()
-                            if (context is Path) {
-                                val fullPath = watchKeys[key]?.resolve(context)
-                                LOG.debug { "$fullPath was changed" }
-                                if (watchLocations.contains(fullPath)) {
-                                    // In case of back to back events, de-dupe them
-                                    invokeListeners = true
-                                }
-                            }
-                        }
-                    }
-                } catch (e: Exception) {
-                    LOG.warn(e) { "ProfileWatcher got an exception" }
-                }
-            }
-
-            val isValid = key.reset()
-            if (!isValid) {
-                LOG.debug { "WatchKey $key is no longer valid" }
-                watchKeys.remove(key)
-            }
-
-            if (watchKeys.isEmpty()) {
-                LOG.debug { "All watch keys have been removed, terminating watcher." }
-                return
-            }
-
-            if (invokeListeners) {
-                listeners.forEach { listener ->
-                    LOG.tryOrNull("Failed to notify listeners") {
-                        listener.onProfilesChanged()
-                    }
-                }
-
-                invokeListeners = false
-            }
-        }
+        watchRoots.addAll(LocalFileSystem.getInstance().addRootsToWatch(watchLocationsStrings, false))
```

Comment:
Looks like you removed the call to `start` further up - so how does this logic actually get hit?

Reply:
I removed the "restart" call when we create the config/cred files since IJ version can handle watching files and dirs that dont exist yet.

We still start it in the factory